feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-serializer.ts

```diff
@@ -93,34 +93,43 @@ export class OccConfiguratorVariantSerializer
     attribute: Configurator.Attribute,
     occAttributes: OccConfigurator.Attribute[]
   ): void {
+    const uiType = attribute.uiType ?? Configurator.UiType.NOT_IMPLEMENTED;
     const targetAttribute: OccConfigurator.Attribute = {
       key: attribute.name,
       name: attribute.name,
       langDepName: attribute.label,
       required: attribute.required,
       retractTriggered: attribute.retractTriggered,
-      type: this.convertCharacteristicType(
-        attribute.uiType ?? Configurator.UiType.NOT_IMPLEMENTED
-      ),
+      type: this.convertCharacteristicType(uiType),
     };
-
-    if (
-      attribute.uiType === Configurator.UiType.DROPDOWN ||
-      attribute.uiType === Configurator.UiType.DROPDOWN_ADDITIONAL_INPUT ||
-      attribute.uiType === Configurator.UiType.RADIOBUTTON ||
-      attribute.uiType === Configurator.UiType.RADIOBUTTON_ADDITIONAL_INPUT ||
-      attribute.uiType === Configurator.UiType.SINGLE_SELECTION_IMAGE
-    ) {
+    const singleValueTypes = [
+      Configurator.UiType.RADIOBUTTON,
+      Configurator.UiType.RADIOBUTTON_ADDITIONAL_INPUT,
+      Configurator.UiType.DROPDOWN_ADDITIONAL_INPUT,
+      Configurator.UiType.DROPDOWN,
+      Configurator.UiType.SINGLE_SELECTION_IMAGE,
+    ];
+
+    const multiValueTypes = [
+      Configurator.UiType.CHECKBOXLIST,
+      Configurator.UiType.CHECKBOX,
+      Configurator.UiType.MULTI_SELECTION_IMAGE,
+    ];
+
+    const inputTypesSetValue = [
+      Configurator.UiType.STRING,
+      Configurator.UiType.SAP_DATE,
+    ];
+
+    const inputTypesSetFormattedValue = [Configurator.UiType.NUMERIC];
+
+    if (singleValueTypes.includes(uiType)) {
       this.retractValue(attribute, targetAttribute);
```

Comment:
confusing method name as it only eventually retracts it....

Reply:
That's true. Still we shouldn't change this (breaking). This PR anyhow is not interested at all in this method, I just needed to refactor because of new sonar rules